### PR TITLE
Implemented Region Enum and updated code and tests to reflect this change

### PR DESCRIPTION
**Pull request by pedromdsduarte**
* Implemented Region enum class
* Implemented get_country_list class method and respective tests
* Updated code to reflect changes

### CODE REVIEW

File: life_expectancy/region.py [hunk 1]
```diff
@@ -3,7 +3,24 @@
...
+        Args:
+            name (_type_): The name of the member being defined.
+            start (_type_): The start value for the Enum; the default is 1.
+            count (_type_): The number of members currently defined, not including this one.
+            last_values (_type_): A list of the previous values.
+
+        Returns:
+            str: The name of the region.
+        """
         return name
```
Comment (MrCordeiro) [inline life_expectancy/region.py:24]: image

File: life_expectancy/region.py [hunk 2]
```diff
@@ -3,7 +3,24 @@
...
+
+        Args:
+            name (_type_): The name of the member being defined.
+            start (_type_): The start value for the Enum; the default is 1.
+            count (_type_): The number of members currently defined, not including this one.
+            last_values (_type_): A list of the previous values.
+
+        Returns:
+            str: The name of the region.
+        """
```
Comment (MrCordeiro) [inline life_expectancy/region.py:23]: image

Comment (MrCordeiro) [inline life_expectancy/region.py:23]: bravo-applause

File: life_expectancy/region.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    RS = auto()
+    RU = auto()
+    SE = auto()
+    SI = auto()
+    SK = auto()
+    SM = auto()
+    TR = auto()
+    UA = auto()
+    UK = auto()
+    XK = auto()
```
Comment (goncalosantiago99) [inline life_expectancy/region.py:90]: Would love to discus this class, it is very interesting.

File: life_expectancy/tests/test_region.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    """
+    country_list = Region.get_country_list()
+
+    # Assert that there are no countries with more than two letters
+    country_list_with_len_gt_2 = [c for c in country_list if len(c) > 2]
+    assert len(country_list_with_len_gt_2) == 0
+
+    # Assert that there are no countries that shouldn't be there
+    not_countries = {"DE_TOT", "EEA30_2007", "EU27_2007"}
+    assert len(set(country_list).intersection(not_countries)) == 0
```
Comment (goncalosantiago99) [inline life_expectancy/tests/test_region.py:18]: Smart to also test the cases where the string doesn't relate to countries. However, I think the test before covers this 3 usecases.

**Review by goncalosantiago99 — APPROVED** [review]
Legend's work! 🥇

**Pull request merged**